src/random/random.c

```diff
@@ -282,30 +487,49 @@ const char *igraph_rng_name(const igraph_rng_t *rng) {
  * O(1).
  */
 
-long int igraph_rng_get_integer(igraph_rng_t *rng,
-                                long int l, long int h) {
-    assert(h >= l);
-    const igraph_rng_type_t *type = rng->type;
+igraph_integer_t igraph_rng_get_integer(
+    igraph_rng_t *rng, igraph_integer_t l, igraph_integer_t h
+) {
+    igraph_uint_t range;
+
     /* We require the random integer to be in the range [l, h]. We do so by
```

Comment:
The comments starting here appear to be outdated, and I think they should be removed.